### PR TITLE
feat: add/select network flow

### PR DESCRIPTION
**Pull request by helciofranco**
- Closes https://github.com/FuelLabs/fuels-wallet/issues/1163

---

| 📷  Demo |
| --- |
| <video src="https://github.com/user-attachments/assets/57d657de-2315-407a-9683-daf002235b78" /> |

### CODE REVIEW

**Comment by luizstacio** [comment]
@helciofranco On the specs of the connectors, we have two different methods for adding and another for selecting a network. https://github.com/FuelLabs/fuel-connectors/wiki.

The behaviors I think would be;
1. Dapp can request add a new network, once this is done it also switches the network. This is done by providing a network URL that does not exist.
2. For know network chainIds (that already exists) the DApp will need to;
2.1 Do a selectNetwork we can allow partial data on this endpoint this means it can receive the chainId == networkId or network URL.
2.2 To verify if they should call a add or select they will use networks that returns the support list of networks inside the connector.

**Review by arthurgeron — APPROVED** [review]

File: packages/app/src/systems/CRX/background/services/BackgroundService.ts [hunk 1]
```diff
@@ -307,13 +309,31 @@ export class BackgroundService {
...
     };
   }
 
+  async networks(): Promise<Network[]> {
+    const networks = await NetworkService.getNetworks();
+    return networks.map((network) => {
+      const isValidChainId = !Number.isNaN(Number(network.id));
+
+      return {
+        chainId: isValidChainId ? Number(network.id) : 0,
```
Comment (LuizAsFight) [inline packages/app/src/systems/CRX/background/services/BackgroundService.ts:331]: what is the use case to have a nan chainId? we should never have it

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
the id being used as chainId is wrong..

we should:
1- add a new field to network table containing `chainId` 
2- every time we add a new network, we save its chainId together


this chainId is the one we need to compare, right now we're comparting to `network.id` which is the primaruy key of network database table, which will be a string key.

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
the id used to compare in the wallet is the `network.id` which is not right because this is a primary key from db table "network", and it will be a key

we should:
1- create a new field `chainId` on networks
2- every time we add a network, we fill the `chainId` together to save in db (it comes already in provider after `Provider.create`)
3- when sending `selectNetwork` passing `chainId` it should compare to this new field in network

File: packages/app/src/systems/CRX/background/services/BackgroundService.ts (hunk 1 above)
Comment (helciofranco) [reply packages/app/src/systems/CRX/background/services/BackgroundService.ts:331]: it was just because i was confusing the `uuid` as the `chainId`
now that we'll add one more column just for that, i'll remove these NaN checks

File: packages/app/src/systems/CRX/background/services/BackgroundService.ts [hunk 2]
```diff
@@ -307,13 +309,31 @@ export class BackgroundService {
     return currentAccount?.address;
   }
 
-  async network() {
+  async network(): Promise<Network> {
     const selectedNetwork = await NetworkService.getSelectedNetwork();
+
+    if (!selectedNetwork) return { chainId: 0, url: '' };
```
[the remark this answers is not in the document]
Comment (helciofranco) [reply packages/app/src/systems/CRX/background/services/BackgroundService.ts:315]: makes sense

File: examples/cra-dapp/package.json [hunk 3]
```diff
@@ -7,8 +7,8 @@
     "start": "vite"
   },
   "dependencies": {
-    "@fuels/connectors": "0.25.0",
-    "@fuels/react": "0.25.0",
+    "@fuels/connectors": "pr-251",
+    "@fuels/react": "pr-251",
```
[the remark this answers is not in the document]
Comment (helciofranco) [reply examples/cra-dapp/package.json:11]: sure, let's finish all releases in the connectors/sdk before updating here

File: .changeset/tough-bobcats-retire.md [hunk 4]
```diff
@@ -2,4 +2,6 @@
 "fuels-wallet": minor
 ---
 
-Allow users to switch to an existing network through the `addNetwork` flow, even if the network already exists in their wallet.
\ No newline at end of file
+Allow users to switch to an existing network through the `selectNetwork` flow.
+It creates the network if it doesn't exist yet.
+It selects the network if it already exists in their wallet.
```
[the remark this answers is not in the document]
Comment (helciofranco) [reply .changeset/tough-bobcats-retire.md:7]: I'm wondering why you suggest that?

I'm referencing this current change (which is "allowing users to switch network"), those lines you mentioned are not the change itself – but an explanation.

```
Allow users to switch to an existing network through the `selectNetwork` flow.
It creates the network if it doesn't exist yet.
It selects the network if it already exists in their wallet.
```

**Comment by helciofranco** [comment]
@LuizAsFight ohhh, makes sense. thanks, i'll create that column for the `chainId`

File: .changeset/tough-bobcats-retire.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (LuizAsFight) [reply .changeset/tough-bobcats-retire.md:7]: because it will look ugly in the changelog. It should be a single sentence then if you want to include all this info, maybe you can reword it to make it smaller

[the remark this answers is not in the document]
Comment (LuizAsFight) [reply .changeset/tough-bobcats-retire.md:7]: we should not break lines in changeset

**Review by arthurgeron — APPROVED** [review]

**Review by LuizAsFight — APPROVED** [review]
